test/multi_process_test.rb

```diff
@@ -7,36 +7,34 @@
     # https://github.com/resque/resque-scheduler/pull/439#discussion_r16788812
     omit('forking is not supported by jruby but this behaviour' \
       ' is best tested using forks') if RUBY_ENGINE == 'jruby'
-    schedules = {}
-    schedules = {}
-    counts = []
+    schedules_1 = {}
+    schedules_2 = {}
+    schedules = []
     pids = []
 
     # This number may need to be increased if this test is not failing
-    processes = 20
+    processes = 100
 
-    schedule_count = 200
+    schedule_count = 300
 
     schedule_count.times do |n|
-      schedules["job #{n}"] = { cron: '0 1 0 0 0' }
+      schedules_1["1_job_#{n}"] = { cron: '0 1 0 0 0' }
+      schedules_2["2_job_#{n}"] = { cron: '0 1 0 0 0' }
     end
 
     processes.times do |n|
       pids << fork_with_marshalled_pipe_and_result do
         sleep n * 0.1
-        Resque.schedule = schedules
-        Resque.schedule.size
+        Resque.schedule = n.even? ? schedules_2 : schedules_1
+        Resque.schedule
       end
     end
 
-    # doing this outside the threads increases the odds of failure
-    Resque.schedule = schedules
-    counts << Resque.schedule.size
-
-    counts += get_results_from_children(pids)
+    schedules += get_results_from_children(pids)
 
-    counts.each_with_index do |c, i|
-      assert_equal schedule_count, c, "schedule count is incorrect (c: #{i})"
+    schedules.each_with_index do |schedule, i|
```

Comment:
Not sure if this makes sense, but what jumps out to me here is that the assertion on the line below might pass (or rather: might never run) if `schedules` is empty. Maybe add a `assert_equal some_number, schedules.size`?